core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java

```diff
@@ -83,6 +90,7 @@ void persistTraits() {
             if (RudderClient.getInstance() != null && RudderClient.getInstance().getApplication() != null) {
                 RudderPreferenceManager preferenceManger = RudderPreferenceManager.getInstance(RudderClient.getInstance().getApplication());
                 preferenceManger.saveTraits(new Gson().toJson(this.traits));
+                preferenceManger.getAnonymousId();
```

Comment:
Are we trying to keep the anonymousId ready before the next call?

Reply:
yes, it is being saved for the next call.